packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -255,6 +255,24 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
 		super.startSummary(referenceSequenceNumber, summaryLogger);
 	}
 
+	protected validateSummaryCore(parentSkipRecursion: boolean) {
+		/**
+		 * The absence of wip used routes indicates that GC was not run on this node. This can happen if:
+		 * 1. A child node was created after GC was already run on the parent. For example, a data store
+		 * is realized (loaded) after GC was run on it creating summarizer nodes for its DDSes. In this
+		 * case, the used routes of the parent should be passed on the child nodes and it should be fine.
+		 * 2. A new node was created but GC was never run on it. This can mean that the GC data generated
+		 * during summarize is complete . We should not continue, log and throw an error. This will help us
+		 * identify these cases and take appropriate action.
+		 */
+		if (!this.gcDisabled && this.wipSerializedUsedRoutes === undefined) {
```

Comment:
Off-topic: Could we simply avoid composing in this sub-class if GC is disabled?

Reply:
Do you mean just have `SummarizerNode`? I guess we could but not sure how much it makes sense seeing as GC is enabled by default.

Reply:
> seeing as GC is enabled by default

That's my point - this class would be cleaner if it could assume that GC is always enabled.  Then whoever creates the nodes can choose which class to use based on whether it actually is enabled.  Would probably require some serious refactoring, it's a different composition pattern.

Reply:
yep, we can create a work item for this under GC code cleanup feature. It shouldn't be hard to do.

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/4726